### PR TITLE
fix: updated company delete

### PR DESCRIPTION
**Pull request by charith-rc**
### Summary

- Correct the company deletion flow by soft deleting only the entity and handle cascading from the get queries themselves

### How to test

1.

### Project Checklist

- [x] Changes build without any errors
- [x] Have written adequate test cases
- [x] Done developer testing in
  - [x] Chrome
  - [ ] Firefox
  - [ ] Safari
- [x] Code is formatted with `npm run format`
- [x] Code is linted with `npm run check-lint`
- [x] No unnecessary comments left in code
- [ ] Made corresponding changes to the documentation

#### Other

- [ ] New atomic components added
- [ ] New molecules added
- [ ] New pages(routes) added
- [ ] New dependencies installed

### PR Checklist

- [x] Pull request is raised from the correct source branch
- [x] Pull request is raised to the correct destination branch
- [x] Pull request is raised with correct title
- [x] Pull request is self reviewed
- [x] Pull request is self assigned
- [ ] Suitable pull request status labels are added (`ready-for-code-review`)

### Additional Information
-

### CODE REVIEW

File: backend/src/main/java/com/skapp/community/crmplanner/repository/impl/CrmContactRepositoryImpl.java [hunk 1]
```diff
@@ -93,6 +93,7 @@ private Predicate[] buildPredicates(CriteriaBuilder cb, Root<CrmContact> contact
 		Long companyId = filterDto.getCompanyId();
 		if (companyId != null) {
 			predicates.add(cb.equal(company.get(CrmCompany_.id), companyId));
+			predicates.add(cb.isFalse(company.get(CrmCompany_.isDeleted)));
 		}
```
[the remark this answers is not in the document]
Comment (charith-rc) [reply backend/src/main/java/com/skapp/community/crmplanner/repository/impl/CrmContactRepositoryImpl.java:95]: fixed to null safe predicate https://github.com/SkappHQ/skapp/pull/1533/commits/2357288c852832d2475f4c0bb8c682215746f020

[the remark this answers is not in the document]
Comment (charith-rc) [reply backend/src/main/java/com/skapp/community/crmplanner/repository/impl/CrmContactRepositoryImpl.java:95]: fixed to null safe predicate https://github.com/SkappHQ/skapp/pull/1533/commits/2357288c852832d2475f4c0bb8c682215746f020

File: backend/src/main/java/com/skapp/community/crmplanner/repository/impl/CrmTaskRepositoryImpl.java [hunk 2]
```diff
@@ -233,6 +233,7 @@ private List<Predicate> buildTaskPredicates(CriteriaBuilder cb, Root<CrmTask> ro
 
 		if (params.getCompanyId() != null) {
 			predicates.add(cb.equal(root.get(CrmTask_.company).get(CrmCompany_.id), params.getCompanyId()));
+			predicates.add(cb.isFalse(root.get(CrmTask_.company).get(CrmCompany_.isDeleted)));
 		}
```
[the remark this answers is not in the document]
Comment (charith-rc) [reply backend/src/main/java/com/skapp/community/crmplanner/repository/impl/CrmTaskRepositoryImpl.java:253]: fixed to null safe predicate https://github.com/SkappHQ/skapp/pull/1533/commits/2357288c852832d2475f4c0bb8c682215746f020

File: backend/src/main/java/com/skapp/community/crmplanner/repository/impl/CrmDealRepositoryImpl.java [hunk 3]
```diff
@@ -122,6 +122,7 @@ private List<Predicate> buildPredicates(CriteriaBuilder cb, Root<CrmDeal> deal,
 
 		if (filterDto.getCompanyId() != null) {
 			predicates.add(cb.equal(deal.get(CrmDeal_.company).get(CrmCompany_.id), filterDto.getCompanyId()));
+			predicates.add(cb.isFalse(deal.get(CrmDeal_.company).get(CrmCompany_.isDeleted)));
 		}
```
[the remark this answers is not in the document]
Comment (charith-rc) [reply backend/src/main/java/com/skapp/community/crmplanner/repository/impl/CrmDealRepositoryImpl.java:128]: fixed to null safe predicate https://github.com/SkappHQ/skapp/pull/1533/commits/2357288c852832d2475f4c0bb8c682215746f020

File: backend/src/main/java/com/skapp/community/crmplanner/repository/impl/CrmDealRepositoryImpl.java [hunk 4]
```diff
@@ -122,6 +122,7 @@ private List<Predicate> buildPredicates(CriteriaBuilder cb, Root<CrmDeal> deal,
...
```
[the remark this answers is not in the document]
Comment (charith-rc) [reply backend/src/main/java/com/skapp/community/crmplanner/repository/impl/CrmDealRepositoryImpl.java:129]: fixed to null safe predicate https://github.com/SkappHQ/skapp/pull/1533/commits/2357288c852832d2475f4c0bb8c682215746f020

File: backend/src/main/java/com/skapp/community/crmplanner/service/impl/CrmCompanyServiceImpl.java [hunk 5]
```diff
@@ -156,10 +152,6 @@ public ResponseEntityDto deleteCompany(Long id) {
 		CrmCompany company = crmCompanyDao.findByIdAndIsDeletedFalse(id)
 			.orElseThrow(() -> new ModuleException(CrmMessageConstant.CRM_ERROR_COMPANY_NOT_FOUND));
 
-		List<CrmDeal> deals = crmDealDao.findAllByCompanyIdAndIsDeletedFalse(id);
-		deals.forEach(deal -> deal.setIsDeleted(true));
-		crmDealDao.saveAll(deals);
-
 		company.setIsDeleted(true);
 		crmCompanyDao.save(company);
```
[the remark this answers is not in the document]
Comment (charith-rc) [reply backend/src/main/java/com/skapp/community/crmplanner/service/impl/CrmCompanyServiceImpl.java:161]: addressed by added predicates https://github.com/SkappHQ/skapp/commit/2357288c852832d2475f4c0bb8c682215746f020

**Review by ThinuwanW — COMMENTED** [review]
## 🤖 Claude Code Review

This PR replaces cascade soft-deletion of deals on company deletion with read-time filtering that excludes contacts/deals/tasks linked to a soft-deleted company, and hardens contact create/edit to reject deleted/non-existent companies. The approach is sound, but the read-filter is applied inconsistently: the main deal list (findDeals) only filters out deleted-company deals when a companyId filter is supplied, so the default deal list still surfaces deals of deleted companies — diverging from the board view, contacts, and tasks. No test was added to cover the new filtering behavior.

Found **6** new issue(s): 🟡 2 suggestion(s)

File: backend/src/main/java/com/skapp/community/crmplanner/repository/impl/CrmDealRepositoryImpl.java [hunk 6]
```diff
@@ -122,6 +123,7 @@ private List<Predicate> buildPredicates(CriteriaBuilder cb, Root<CrmDeal> deal,
 
 		if (filterDto.getCompanyId() != null) {
 			predicates.add(cb.equal(deal.get(CrmDeal_.company).get(CrmCompany_.id), filterDto.getCompanyId()));
+			predicates.add(cb.isFalse(deal.get(CrmDeal_.company).get(CrmCompany_.isDeleted)));
```
Comment (ThinuwanW) [inline backend/src/main/java/com/skapp/community/crmplanner/repository/impl/CrmDealRepositoryImpl.java]: 🤖 **Claude** · 💬 **Note**

Inconsistent deleted-company filtering. Here the deleted-company predicate is only added inside the `if (filterDto.getCompanyId() != null)` block, so the default deal list (no companyId filter) still returns deals belonging to soft-deleted companies. Compare buildStagePredicates (lines 139-140) and countDealsByStageIds (lines 212-213), which always add `cb.or(cb.isNull(companyJoin), cb.isFalse(companyJoin.get(CrmCompany_.isDeleted)))` via a LEFT join. Since this PR removed the cascade soft-delete of deals on company deletion, this list query (findDeals → CrmDealServiceImpl.getDeals) will now leak orphaned deals that the board view correctly hides. Fix: add an unconditional `Join<CrmDeal,CrmCompany> companyJoin = deal.join(CrmDeal_.company, JoinType.LEFT)` with the isNull-or-isFalse(isDeleted) predicate at the top of buildPredicates (and reuse companyJoin for the companyId equality), matching the other two methods. Note `deal.company` is nullable, so a LEFT join with the isNull branch is required — the current `deal.get(CrmDeal_.company).get(...)` implicit inner-join form would also wrongly drop deals with no company if reused outside the companyId block.

File: backend/src/main/java/com/skapp/community/crmplanner/repository/impl/CrmDealRepositoryImpl.java [hunk 7]
```diff
@@ -1,5 +1,6 @@
 package com.skapp.community.crmplanner.repository.impl;
 
+import com.skapp.community.crmplanner.model.CrmCompany;
```
Comment (ThinuwanW) [inline backend/src/main/java/com/skapp/community/crmplanner/repository/impl/CrmDealRepositoryImpl.java:3]: 🤖 **Claude** · 💬 **Note**

📁 **File-level observation**

Asymmetry in contact-scoped queries (suggestion). The list/board queries now exclude records tied to a deleted company, but the contact-detail and metric queries were not updated: CrmDealRepositoryImpl.findByContactIdWithAssociations / findDealMetricsByContactId / findClosedDealSummaryByContactIds, and CrmTaskRepositoryImpl.findByContactIdWithAssociations / findTaskMetricsByContactId / findOpenTaskSummaryByContactIds / countTasksByDealIds still include deals/tasks whose company is soft-deleted. As a result the contact-detail view and its revenue/task counts will include records that no longer appear in any list. If hiding deleted-company records is the intended product behavior, these queries should apply the same `isNull(company) OR isFalse(company.isDeleted)` filter for consistency; if it's intentional to keep them in the detail view, please confirm so the divergence is deliberate.

File: backend/src/test/java/com/skapp/community/crmplanner/controller/v1/CrmCompanyControllerIntegrationTest.java [hunk 8]
```diff
@@ -1,15 +1,6 @@
 package com.skapp.community.crmplanner.controller.v1;
```
Comment (ThinuwanW) [inline backend/src/test/java/com/skapp/community/crmplanner/controller/v1/CrmCompanyControllerIntegrationTest.java:1]: 🤖 **Claude** · 💬 **Note**

📁 **File-level observation**

Missing test coverage for the new behavior. The removed test `deleteCompany_WithAssociatedDeals_SoftDeletesDeals` correctly went away with the cascade, but no replacement test was added to verify the new contract: after a company is soft-deleted its deals are NOT soft-deleted, and its deals/contacts/tasks are excluded from the list/board queries. Per the review guidelines (service/repository changes should have corresponding tests), please add coverage asserting (a) deleting a company leaves its deals' isDeleted=false, and (b) those deals/contacts/tasks no longer appear in the respective list endpoints.

File: backend/src/main/java/com/skapp/community/crmplanner/repository/impl/CrmContactRepositoryImpl.java [hunk 9]
```diff
@@ -145,6 +149,9 @@ private List<Predicate> buildLookupPredicates(CriteriaBuilder cb, Root<CrmContac
 		List<Predicate> predicates = new ArrayList<>();
 		predicates.add(cb.isFalse(contact.get(CrmContact_.isDeleted)));
 
+		Join<CrmContact, CrmCompany> companyJoin = contact.join(CrmContact_.company, JoinType.LEFT);
```
Comment (ThinuwanW) [inline backend/src/main/java/com/skapp/community/crmplanner/repository/impl/CrmContactRepositoryImpl.java]: 🤖 **Claude** · 💬 **Note**

Redundant duplicate join (nit). In findContactsForLookup the company association is fetched at line 134 (`contact.fetch(CrmContact_.company, JoinType.LEFT)`) and then joined again here at line 152 inside buildLookupPredicates, producing two separate LEFT joins to the same company table in the generated SQL. findContacts avoids this by casting the fetch to a Join and reusing it (line 55). Functionally correct but slightly wasteful; consider reusing a single join for both the fetch and the predicate, as done in findContacts.

File: backend/src/main/java/com/skapp/community/crmplanner/service/impl/CrmCompanyServiceImpl.java [hunk 10]
```diff
@@ -21,9 +21,7 @@
 import com.skapp.community.crmplanner.payload.response.CrmNameExistsResponseDto;
```
Comment (ThinuwanW) [inline backend/src/main/java/com/skapp/community/crmplanner/service/impl/CrmCompanyServiceImpl.java:21]: 🤖 **Claude** · 🟡 **Suggestion**

📁 **File-level observation**

Dead code left by removing the cascade. The deleteCompany cascade was the only caller of CrmDealDao.findAllByCompanyIdAndIsDeletedFalse(Long), and a codebase-wide search shows it is now unused (only declared in CrmDealDao.java:19, referenced nowhere in main or test). Remove the now-orphaned method from CrmDealDao to avoid leaving dead query methods behind. (Distinct from the first-pass filtering/asymmetry findings.)

File: backend/src/main/java/com/skapp/community/crmplanner/repository/impl/CrmDealRepositoryImpl.java [hunk 11]
```diff
@@ -134,6 +136,9 @@ private List<Predicate> buildStagePredicates(CriteriaBuilder cb, Root<CrmDeal> d
 		predicates.add(cb.equal(deal.get(CrmDeal_.stage).get(CrmDealStage_.id), stageId));
 		predicates.add(cb.isFalse(deal.get(CrmDeal_.isDeleted)));
 
+		Join<CrmDeal, CrmCompany> companyJoin = deal.join(CrmDeal_.company, JoinType.LEFT);
+		predicates.add(cb.or(cb.isNull(companyJoin), cb.isFalse(companyJoin.get(CrmCompany_.isDeleted))));
```
Comment (ThinuwanW) [inline backend/src/main/java/com/skapp/community/crmplanner/repository/impl/CrmDealRepositoryImpl.java]: 🤖 **Claude** · 🟡 **Suggestion**

Code duplication of the not-deleted-company guard. The predicate `cb.or(cb.isNull(companyJoin), cb.isFalse(companyJoin.get(CrmCompany_.isDeleted)))` (with its preceding LEFT join to CrmCompany) is now copy-pasted in at least five places across three files: CrmDealRepositoryImpl.buildStagePredicates (140) and countDealsByStageIds (213), CrmContactRepositoryImpl.buildPredicates (98), findAllContactsForBoardInit (123) and buildLookupPredicates (153), and CrmTaskRepositoryImpl.buildTaskPredicates (208). Consider extracting a shared helper (e.g. a small static utility taking cb + the company Join and returning the Predicate) so the rule lives in one place. This also reduces the risk that future edits update some call sites but not others — which is exactly the kind of inconsistency the first pass already flagged in CrmDealRepositoryImpl.buildPredicates.

Comment (charith-rc) [reply backend/src/main/java/com/skapp/community/crmplanner/repository/impl/CrmDealRepositoryImpl.java]: extracted a query util to reuse at all call sites

File: backend/src/main/java/com/skapp/community/crmplanner/service/impl/CrmCompanyServiceImpl.java (hunk 10 above)
Comment (charith-rc) [reply backend/src/main/java/com/skapp/community/crmplanner/service/impl/CrmCompanyServiceImpl.java:21]: removed CrmDealDao.findAllByCompanyIdAndIsDeletedFalse

File: backend/src/test/java/com/skapp/community/crmplanner/controller/v1/CrmCompanyControllerIntegrationTest.java (hunk 8 above)
Comment (charith-rc) [reply backend/src/test/java/com/skapp/community/crmplanner/controller/v1/CrmCompanyControllerIntegrationTest.java:1]: added test coverage to cover edge cases

File: backend/src/main/java/com/skapp/community/crmplanner/repository/impl/CrmDealRepositoryImpl.java (hunk 6 above)
Comment (charith-rc) [reply backend/src/main/java/com/skapp/community/crmplanner/repository/impl/CrmDealRepositoryImpl.java]: fixed by adding a left join with the isFalse(isDeleted) predicate to filter out deleted companies in CrmDealRepositoryImpl

File: backend/src/main/java/com/skapp/community/crmplanner/repository/impl/CrmDealRepositoryImpl.java (hunk 7 above)
Comment (charith-rc) [reply backend/src/main/java/com/skapp/community/crmplanner/repository/impl/CrmDealRepositoryImpl.java:3]: applied same guard to deals and tasks

File: backend/src/main/java/com/skapp/community/crmplanner/repository/impl/CrmContactRepositoryImpl.java (hunk 9 above)
Comment (charith-rc) [reply backend/src/main/java/com/skapp/community/crmplanner/repository/impl/CrmContactRepositoryImpl.java]: added the query util here as well to remove duplication

File: backend/src/main/java/com/skapp/community/crmplanner/repository/impl/CrmDealRepositoryImpl.java (hunk 11 above)
Comment (charith-rc) [reply backend/src/main/java/com/skapp/community/crmplanner/repository/impl/CrmDealRepositoryImpl.java]: reverted this back to query level. extracting a shared helper for a predicate means we couple a predicate that holds a responsibility in many queries to one site. any modifications needed to one query would break other call sites. since this case is a criteria query, extraction to a util may not be the best way.

**Review by akilarootcode — APPROVED** [review]

**Review by ThinuwanW — APPROVED** [review]
